src/modules/ganalytics/AnalyticsProvider.tsx

```diff
@@ -1,107 +1,177 @@
+}
 
-export function AnalyticsProvider(props: { children: React.ReactNode }) {
-  // @ts-ignore
-  let gtag: any;
-  let solPrice = 0;
+export const ga4Event = (
+  action: Gtag.EventNames | AnalyticsAction,
+  {
+    event_category,
+    event_label,
+    value,
+    page_path,
+    ...otherAttributes
+  }: Gtag.EventParams & { page_path: string; [key: string]: any }
+) => {
+  window.gtag('event', action, {
+    event_category,
+    event_label,
+    value,
+    page_path,
+    ...otherAttributes,
+  });
+};
 
-  //   const endpointName = ENDPOINTS.find((e) => e.endpoint === endpoint)?.name;
+interface IAnalyticsContext {
+  track: (action: AnalyticsAction, attributes: TrackingAttributes) => void;
+}
+
+const AnalyticsContext = React.createContext<IAnalyticsContext | null>(null);
+
+export function AnalyticsProvider(props: { children: React.ReactNode }) {
+  const router = useRouter();
+  const [trackingInitialized, setTrackingInitialized] = useState(true);
+  const [trackingAccepted, setTrackingAccepted] = useState(true);
   const { wallet } = useContext(WalletContext);
   const pubkey = wallet?.pubkey || '';
   // const pubkey = publicKey?.toBase58() || '';
-  useEffect(() => {
-    return
-    gtag = window?.gtag;
-    // const isStoreOwner = ownerAddress === publicKey?.toBase58();
-    // user pubkey / id
-
-    setUserProperties({
-      user_id: pubkey,
-      pubkey: pubkey,
-    });
+  // const endpointName = ENDPOINTS.find((e) => e.endpoint === endpoint)?.name;
+
+  let solPrice = 0;
+
+  function initializeTracking() {
+    console.log('Initialize analytics');
     new Coingecko().getRate([Currency.SOL], Currency.USD).then((rates) => {
       const solRate = rates[0].rate;
       solPrice = solRate;
     });
 
-    try {
-    } catch (error) {}
+    if (GA4_ID) {
+      const sendPageView = true;
+      gtag('config', GA4_ID, {
+        send_page_view: sendPageView,
+      });
+      if (sendPageView) {
+        console.log('Initial google pageview');
+      }
+      setTrackingInitialized(true);
+    }
 
-    // initial config
-    configureAnalytics({
-      //   network: endpointName,
-    });
-  }, [
-    pubkey,
-    // endpointName
-  ]);
+    if (BUGSNAG_API_KEY) {
+      Bugsnag.start({
+        appVersion: '0.1.0', // TODO: Link to app version
+        apiKey: BUGSNAG_API_KEY,
+        plugins: [new BugsnagPluginReact()],
+        onError(event) {
+          if (pubkey) {
+            event.setUser(pubkey);
+          }
+          if (false) {
+            // for later inspiration
+            event.addMetadata('company', {
+              name: 'Acme Co.',
+              country: 'uk',
+            });
+          }
+        },
+      });
+    }
+  }
 
-  function setUserProperties(attributes: AnalyticsUserProperties) {
-    gtag('set', 'user_properties', {
-      ...attributes,
-    });
+  function identify() {
+    console.log(pubkey ? 'User identified' : 'No user identified');
```

Comment:
remove. We should add no console log to our linting.

Reply:
Got it, I'll clean up all the console logs :)